akka-http-core/src/main/scala/akka/http/impl/engine/http2/hpack/HeaderCompression.scala

```diff
@@ -44,7 +44,7 @@ private[http2] object HeaderCompression extends GraphStage[FlowShape[FrameEvent,
         case ParsedHeadersFrame(streamId, endStream, kvs, prioInfo) =>
           kvs.foreach {
             case (key, value) =>
-              encoder.encodeHeader(os, key.getBytes(HeaderDecompression.UTF8), value.getBytes(HeaderDecompression.UTF8), false)
+              encoder.encodeHeader(os, key.getBytes(HeaderDecompression.US_ASCII), value.getBytes(HeaderDecompression.US_ASCII), false)
```

Comment:
for ascii it's possible to take it one step further by using this in Akka:
https://github.com/akka/akka/blob/206dafa01d81224e5f93ddc6e6ec7fa702a03ce3/akka-actor/src/main/scala/akka/util/Unsafe.java#L129

Reply:
Ah, interesting, I think we have a similar helper already also in akka-http as well.